gapic/src/main/com/google/gapid/views/AtomTree.java

```diff
@@ -450,7 +455,13 @@ public void updateElement(Object parent, int index) {
       AtomStream.Node child = ((AtomStream.Node)parent).getChild(index);
       atoms.load(child, refresher::refresh);
       viewer.replace(parent, index, child);
-      viewer.setChildCount(child, child.getChildCount());
+      if (child.getChildCount() > 100) {
+        // For nodes with lots of children, only allocate the children, once the node
+        // is actually expanded.
+        viewer.setHasChildren(child, true);
```

Comment:
Is this notably slower / less efficient than `setChildCount`? Just wondering why you need to put this on a conditional?
Also, I assume using `setChildCount` doesn't cause the client to ask the counts of all the children before they're expanded?